kernel/Kconfig

```diff
@@ -561,6 +561,22 @@ config USE_SWITCH
 	architecture provides both, _arch_switch incurs more somewhat
 	overhead and may be slower.
 
+config SMP
+	bool
+	prompt "Enable symmetric multithreading support"
+	default n
+	help
+	When true, Zephyr will be build with SMP support, allowing
+	more than one CPU to schedule Zephyr tasks at a time.
+
+config MP_NUM_CPUS
+	int
+	prompt "Number of CPUs/cores"
+	default 1
```

Comment:
might want to impose a constraint that the max value here is 1 unless CONFIG_SMP is enabled.
should enforce that this can't be less than 1.

Reply:
This is a hardware thing.  You can have more than one CPU available (to the MP startup API, which is kinda documented) but still build a non-SMP Zephyr that ignores them.